core/modules/image/image.field.inc

```diff
@@ -542,9 +556,54 @@ function image_field_widget_process($element, &$form_state, $form) {
     '#access' => (bool) $item['fid'] && $settings['title_field'],
   );
 
+  // Pass alt and title required status to JS in case other modules need access.
+  $element['#attached']['js'][] = array(
+    'type' => 'setting',
+    'data' => array(
+      'image' => array(
+        'altRequired' => $instance['settings']['alt_field_required'],
+      ),
+    ),
+  );
+
+  if ($settings['alt_field_required']) {
+    $element['alt']['#title'] .= ' <span class="form-required" title="' . t('The alt text field is required.') . '">*</span>';
+    $element['alt']['#element_validate'] = array('_image_required_fields_validate');
+  }
+
   return $element;
 }
 
+/**
+ * Validate callback for alt field.
+ *
+ * This is separated in a validate function instead of a #required flag to avoid
+ * being validated on the process callback.
+ */
+function _image_required_fields_validate($element, &$form_state) {
+  // Only do validation if the function is triggered from other places than
+  // the image process form.
+  if (isset($form_state['triggering_element']['#submit']) && !in_array('file_managed_file_submit', $form_state['triggering_element']['#submit'])) {
+    // If the image is not there, we do not check for empty values.
+    $parents = $element['#parents'];
+    $field = array_pop($parents);
+    $image_field = backdrop_array_get_nested_value($form_state['input'], $parents);
+    if ($image_field == NULL) {
+      $image_field = array();
+    }
+    // We check for the array key, so that it can be NULL (like if the user
+    // submits the form without using the "upload" button).
+    if (!array_key_exists($field, $image_field)) {
+      return;
+    }
+    // Check if field is left empty.
+    elseif (empty($image_field[$field])) {
+      form_error($element, t('The !title field is required', array('!title' => ucwords($field))));
+      return;
+    }
+  }
+}
+
```

Comment:
```suggestion

/**
 * Pre-render callback for alt field.
 *
 * Mark a field as #required for display only.
 */
function _image_field_widget_alt_pre_render($element) {
  $element['#required'] = TRUE;
  return $element;
}

```

Reply:
Done with [`7647357` (#4422)](https://github.com/backdrop/backdrop/pull/4422/commits/7647357614d5b91f84f34312a417ba1118344b60) and docblock fixed to match this suggestion with [`98c26c6` (#4422)](https://github.com/backdrop/backdrop/pull/4422/commits/98c26c6a7d9d7cf650d84329b23f567d521727eb)